core/internal/monitor/monitor.go

```diff
@@ -363,6 +347,31 @@ func (sm *SystemMonitor) Start(git *spb.GitRepoRecord) {
 	}
 }
 
+// Probe collects environment and resource information.
+//
+// This operation may take some time, so we perform it on a best-effort basis.
+func (sm *SystemMonitor) Probe() {
+	if !sm.settings.IsDisableMeta() && !sm.settings.IsDisableMachineInfo() {
```

Comment:
Does this not need the `sm.settings.IsPrimary()` check?

Reply:
Not anymore, but it's a separate story - I'll remove it from here and put in another PR.